File: AspNetMvcUnleashed/0672329980_source/Chapter12/CS/MvcApplication1/MvcApplication1.Tests/Controllers/JackControllerTests.cs
using System.Web.Mvc;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using MvcApplication1.Controllers;

namespace MvcApplication1.Tests.Controllers
{
    [TestClass]
    public class JackControllerTests
    {

        [TestMethod]
        public void JackCanAccessIndex()
        {
            // Arrange
            var controller = new CompanyController();
            var indexAction = typeof(JackController).GetMethod("Index");
            var authorizeAttributes=indexAction.GetCustomAttributes(typeof(AuthorizeAttribute), true);

            // Assert
            Assert.IsTrue(authorizeAttributes.Length > 0);
            foreach (AuthorizeAttribute att in authorizeAttributes)
                Assert.AreEqual("Jack", att.Users);
        }
    }
}

File: AspNetMvcUnleashed/0672329980_source/Chapter12/VB/MvcApplication1/MvcApplication1.Tests/Controllers/JackControllerTests.vb
Imports Microsoft.VisualStudio.TestTools.UnitTesting
Imports System.Web.Mvc

<TestClass()> _
Public Class JackControllerTests

    <TestMethod()> _
    Public Sub JackCanAccessIndex()
        ' Arrange
        Dim controller As New CompanyController()
        Dim indexAction = GetType(JackController).GetMethod("Index")
        Dim authorizeAttributes = indexAction.GetCustomAttributes(GetType(AuthorizeAttribute), True)

        ' Assert
        Assert.IsTrue(authorizeAttributes.Length > 0)
        For Each att As AuthorizeAttribute In authorizeAttributes
            Assert.AreEqual("Jack", att.Users)
        Next att
    End Sub
End Class
